### PR TITLE
Update offset when party/config rejected on a duplicate

### PR DESCRIPTION
**Pull request by mziolekda**
CHANGELOG_BEGIN
Advance ledger end when the indexer encounters an duplicate party or configuration update. This allows indexer to skip-over the unexpected message and continue processing subsequent updates.
CHANGELOG_END

### CODE REVIEW

**Review by rautenrieth-da — APPROVED** [review]
The fix and the new tests looks good.

I think at some point we should move away from the current approach (unconditional insert + scan the error message) to something that does not rely on SQL errors. For example, add `ON CONFLICT ... DO NOTHING` to the insert statement and check the number of written rows to determine whether to emit a log warning.

**Review by tudor-da — APPROVED** [review]

**Comment by oliverse-da** [comment]
Thank you, @mziolekda !

File: ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoPartiesSpec.scala [hunk 1]
```diff
@@ -123,18 +123,47 @@ private[dao] trait JdbcLedgerDaoPartiesSpec {
...
+  it should "store just offset when duplicate party update encountered" in {
+    val party = Ref.Party.assertFromString(s"Alice-${UUID.randomUUID()}")
+    val aliceDetails = PartyDetails(
+      party,
+      displayName = Some("Alice Arkwright"),
+      isLocal = true,
+    )
+    val aliceAgain = PartyDetails(
+      party,
+      displayName = Some("Alice Carthwright"),
```
Comment (miklos-da) [inline ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoPartiesSpec.scala:135]: Does the `displayName` matter from the test case's perspective? If not, you could simply assign `aliceAgain` to `aliceDetails`.

Comment (rautenrieth-da) [reply ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoPartiesSpec.scala:135]: It does, the test asserts that the second `storePartyEntry` call did not alter the already existing entry (see `parties should contain allOf (aliceDetails, ...)`. We could have decided to update existing entries instead.

**Pull request merged**